test/libsinsp_e2e/event_capture.cpp

```diff
@@ -24,208 +24,126 @@ limitations under the License.
+	}
 
-	ASSERT_FALSE(get_inspector()->is_capture());
-	ASSERT_FALSE(get_inspector()->is_live());
-	ASSERT_FALSE(get_inspector()->is_nodriver());
+	m_before_capture(m_inspector.get());
+	m_inspector->start_capture();
+}
 
-	try {
-		if(m_mode == SINSP_MODE_NODRIVER) {
-			get_inspector()->open_nodriver();
-		} else {
-			open_engine(event_capture::get_engine(), {});
-		}
-	} catch(sinsp_exception& e) {
-		m_start_failed = true;
-		m_start_failure_message =
-		        "couldn't open inspector (maybe driver hasn't been loaded yet?) err=" +
-		        get_inspector()->getlasterr() + " exception=" + e.what();
-		{
-			m_capture_started = true;
-			m_condition_started.notify_one();
-		}
-		return;
+void event_capture::stop() {
+	close(m_eventfd);
+	m_inspector->stop_capture();
+	m_after_capture(m_inspector.get());
+	if(m_dumper != nullptr) {
+		m_dumper->close();
 	}
-
-	get_inspector()->set_debug_mode(true);
-	get_inspector()->set_hostname_and_port_resolution_mode(false);
+	m_inspector->close();
 }
 
 void event_capture::capture() {
-	const ::testing::TestInfo* const test_info =
-	        ::testing::UnitTest::GetInstance()->current_test_info();
-	std::unique_ptr<sinsp_cycledumper> dumper;
-	{
-		std::scoped_lock init_lock(m_inspector_mutex, m_object_state_mutex);
-
-		if(!inspector_ok) {
-			init_inspector();
-			if(!m_start_failed) {
-				inspector_ok = true;
-			} else {
-				std::cerr << m_start_failure_message << std::endl;
-				return;
-			}
-		}
-
-		m_param.m_inspector = get_inspector();
-
-		m_before_open(get_inspector());
-
-		get_inspector()->start_capture();
-		if(m_mode != SINSP_MODE_NODRIVER) {
-			m_dump_filename = std::string(LIBSINSP_TEST_CAPTURES_PATH) +
-			                  test_info->test_case_name() + "_" + test_info->name() + ".scap";
-			dumper = std::make_unique<sinsp_cycledumper>(get_inspector(),
-			                                             m_dump_filename.c_str(),
-			                                             0,
-			                                             0,
-			                                             0,
-			                                             0,
-			                                             true);
-		}
-	}  // End init synchronized section
-
-	bool signaled_start = false;
 	sinsp_evt* event;
 	bool result = true;
 	int32_t next_result = SCAP_SUCCESS;
+
+	/*
+	 * Loop until:
+	 * * test has non fatal failures
+	 * * handle_event returns true
+	 * * we weren't asked to leave from eventfd and we received !SCAP_SUCCESS
+	 */
 	while(result && !::testing::Test::HasFatalFailure()) {
-		{
-			std::lock_guard<std::mutex> lock(m_object_state_mutex);
-			if(m_capture_stopped) {
-				break;
-			}
-		}
-		{
-			std::scoped_lock inspector_next_lock(m_inspector_mutex);
-			next_result = get_inspector()->next(&event);
-		}
-		if(SCAP_SUCCESS == next_result) {
-			result = handle_event(event);
-			if(m_mode != SINSP_MODE_NODRIVER && m_dump) {
-				dumper->dump(event);
-			}
+		handle_eventfd_request();
```

Comment:
Before doing anything else, check if any eventfd request was sent; this is non-blocking of course.

Reply:
If you are wondering why i did not just use eg:
`close(MY_FAKE_FD)` to signal the capture to end, is because during my tests it happened that the signaling event got lost and then :infinity:

Reply:
Of course, having a `read` on each loop creates a small backpressure on the drivers.
But tests that generate lots of syscalls traffic are now enabling only the sinsp state sc set of syscalls, therefore in that case, no `read` will appear in our syscall event loop ;)

Reply:
Latest changes put, by default, the whole sc set minus `read` and `readv` to avoid the backpressure (ie: we do not push eventfd read related events to userspace).
Of course some tests rely on reads, in that case they must explicitly enable the `libsinsp::events::all_sc_set()`.

Also, the sc set is now a parameter of `event_capture::run()` APIs.